src/Rules/Documentation.hs

```diff
@@ -10,6 +10,81 @@ import Settings
 import Target
 import Utilities
 
+docPackage :: Package
+docPackage = hsLibrary "Documentation" "docs"
+
+-- | Build all documentation
+documentationRules :: Rules ()
+documentationRules = do
+    buildHtmlDocumentation
+    buildPdfDocumentation
+    buildDocumentationArchives
+    "docs" ~> do
+        root <- buildRoot
+        -- Archives `need` the associated html
+        let archives = map pathArchive docPaths
+            pdfs = map pathPdf $ drop 1 docPaths
+        need $ map (root -/-) $ archives ++ pdfs
+
+docPaths :: [FilePath]
+docPaths = [ "libraries", "users_guide", "Haddock" ]
+
+docRoot :: FilePath
+docRoot = "docs"
+
+htmlRoot :: FilePath
+htmlRoot = docRoot -/- "html"
+
+pdfRoot :: FilePath
+pdfRoot = docRoot -/- "pdfs"
+
+archiveRoot :: FilePath
+archiveRoot = docRoot -/- "archives"
+
+pathPdf :: FilePath -> FilePath
+pathPdf path = pdfRoot -/- path <.> ".pdf"
+
+pathIndex :: FilePath -> FilePath
+pathIndex path = htmlRoot -/- path -/- "index.html"
+
+pathLibIndex :: FilePath -> FilePath
+pathLibIndex path = htmlRoot -/- "libraries" -/- path -/- "index.html"
+
+pathArchive :: FilePath -> FilePath
+pathArchive path = archiveRoot -/- path <.> "html.tar.xz"
+
+pathPath :: FilePath -> FilePath
+pathPath "users_guide" = "docs/users_guide"
+pathPath "Haddock" = "utils/haddock/doc"
+pathPath _ = "lol"
+
+----------------------------------------------------------------------
+-- HTML
+
+-- | Build all HTML documentation
+buildHtmlDocumentation :: Rules ()
+buildHtmlDocumentation = do
+    mapM_ buildSphinxHtml $ drop 1 docPaths
+    buildLibraryDocumentation
+    "//html/index.html" %> \file -> do
+        root <- buildRoot
+        need $ map ((root -/-) . pathIndex) docPaths
+        copyFileUntracked "docs/index.html" file
```

Comment:
Why do we need to keep two copies of the index in the tree?

Reply:
And why is the copy untracked?

Reply:
The file is auto-generated from docs/index.html.in, and so is set from configure time. As it happens, the ticket above asked for some upgrades for that page. After I get to those, this might be a static file that will be easier to handle.